### ISSUE TITLE
Fiddling with jack_link (Ableton Link) & supercollider gives me a oscillating tone

### ISSUE THREAD

**Issue by grammoboy2** [supercollider/supercollider#5284]
Problem: 
Fiddling with jack_link & supercollider (SC) gives me a oscillating tone.

That means, when I experiment with tempo changes via Link from SC, I get a low tone in my speakers, which only disappears when I completely stop SuperCollider. F12 (Cmd+.) doesn't safe me. 

I'm pretty sure I only run into this issue when jack_link AND jack_transport are running. No other JACK apps are running.

I've also reported this to jack_link: https://github.com/rncbc/jack_link/issues/6
https://github.com/rncbc/jack_link

## Environment

* SuperCollider version: 3.11.2 (rc?)
* Operating system: Linux (Debian Testing)
* scnvim

## Steps to reproduce

- Start JACK & Qjackctl
- Make a LinkClock with GUI in SuperCollider (example helpfile)
- Start jack_link
https://github.com/rncbc/jack_link
- Start Jack_transport using the Qjackctl GUI
- Start a simple sound pattern in SuperCollider
- Change tempo in the LinkClock GUI, testing around, 100 bmp, 200, 20, 60....

## Expected vs. actual behavior

I don't experience this tone when I use the same setup with PureData.
I'm only experience this tone when jack_transport and jack_link are running
I don't experience this tone when using SuperCollider normally.

**Comment by grammoboy2** [comment]
Here is more info how to use jack_link by the way: https://linuxmusicians.com/viewtopic.php?f=19&t=22430

**Comment by jamshark70** [comment]
There are a couple of things going on here:

1. "Experimenting with tempo changes" is probably the first thing everyone wants to do with Link but *it's impossible to do it in a completely stable way*. So if you're making rapid, extreme tempo changes, you should expect problems.
   The reason is server messing latency. Our clock runs "early" -- when we reach a downbeat, it's `latency` seconds before the shared downbeat. The tempo change introduces a discontinuity in the timeline. Events that fired within the latency gap will awaken at the wrong time, introducing a risk of stuck nodes. When I repeated your test, wildly changing the tempo, I had 35 stuck nodes in a matter of seconds.
   TL;DR what you were doing -- don't.
   This much, at least, is not a bug. There's nothing we can do to make it perfect. Link is meant as a sync method but it will never be safe for arbitrary, large tempo jumps.
2. https://github.com/supercollider-quarks/JITLibExtensions/blob/main/classes/GUI/ProxyMeter/ProxyMeter.sc#L38 breaks cmd-dot if `TempoClock.default` is not marked `permanent_(true)`. This would prevent cmd-dot from stopping the nodes and would explain why SC is producing a tone that can't be stopped without quitting the server. -- (though you might try `s.defaultGroup.freeAll`)
   I'll file a bug for JITLibExtensions.

I think there's nothing the SC main repository can do to fix this, so I suggest to close the issue.

**Comment by jamshark70** [comment]
Here's a concrete example:

```supercollider
// 1. start JACK
// 2. run jack_link in terminal
// 3. start JACK transport in qjackctl

t = LinkClock.new.latency_(0.2);

t.tempo  // now should be 2 (JACK transport starts at 120 bpm)

f = { |... args| args.debug("LinkClock update") };
t.addDependant(f);

(
r = Routine {
	// next whole beat, schedule a tempo change
	thisThread.clock.sched(1.0, {
		t.tempo = 1;  // 60 bpm
	});
	100.do { |i|
		[i, thisThread.clock.beats, SystemClock.seconds].postln;
		0.05.wait;
	};
}.play(t, 1);
)

t.stop;
```

Output:

```
-> a Routine
[ 0, 19.0, 502.455778 ]
[ 1, 19.05, 502.480778 ]
[ 2, 19.1, 502.505778 ]
... 0.05 beats = 0.025 seconds between lines...
[ 18, 19.9, 502.905778 ]
[ 19, 19.95, 502.930778 ]
LinkClock update: [ a LinkClock, tempo ]
[ 20, 20.0, 502.955778 ]
LinkClock update: [ a LinkClock, tempo ]
[ 21, 20.05, 503.005778 ]
[ 22, 20.1, 505.452393 ]  <<-- 2.446615 sec between these!
[ 23, 20.15, 505.502393 ]
[ 24, 20.2, 505.552393 ]
... and thereafter, 0.05 beats = 0.05 seconds between lines...
```

The 2.446615 seems to be variable -- in some iterations of this test, it was even a negative delta. (This is weird -- I wouldn't have predicted that -- maybe worth looking into later.)

But it does underscore that extreme tempo changes are not a good idea. Gradual ones should be close enough.
